Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -43,42 +47,143 @@ void ScrollAnchoringController::invalidateAnchorElement()
     }
 }
 
-ScrollOffset ScrollAnchoringController::computeOffset(RenderBox& candidate)
+ScrollOffset ScrollAnchoringController::computeOffset(RenderObject& candidate)
 {
     // TODO: investigate this for zoom/rtl
     // make sure this works for subscrollers
     return ScrollOffset(FloatPoint(candidate.absoluteBoundingBoxRect().location() - m_frameView.layoutViewportRect().location()));
 }
 
-void ScrollAnchoringController::chooseAnchorElement()
+static RefPtr<Element> anchorElementForPriorityCandidate(Element* element)
 {
-    // TODO: fully implement anchor node selection algorithm in spec
-    // (specifically excluded subtrees and priority nodes)
-    // https://drafts.csswg.org/css-scroll-anchoring/#anchor-node-selection
+    while (element) {
+        if (auto renderer = element->renderer()) {
+            if (!renderer->isAnonymousBlock() && (!renderer->isInline() || renderer->isAtomicInlineLevelBox()))
+                return element;
+        }
+        element = element->parentElement();
+    }
+    return nullptr;
+}
 
-    auto* document = m_frameView.frame().document();
-    if (!document)
-        return;
+static RefPtr<Element> viablePriorityCandidateForElement(Element* element)
```

Comment:
If you have a static function that's just used at one call site, consider just making it a lambda.

Reply:
Fixed.